src/lib/snack-bar/base-snack-bar.ts

```diff
@@ -0,0 +1,12 @@
+import {MdSnackBarRef} from './snack-bar-ref';
+
+
+export class BaseSnackBarContent<T> {
```

Comment:
I don't want to require that any component you open in the sidenav inherits / implements anything. Inheritance is especially problematic because your component might already inherit from something else. 

All of the end-developer's interactions with the snackbar should be done through the SnackBarRef